### PR TITLE
Add support for OKEX/ExChain

### PR DESCRIPTION
**Pull request by samsondav**

### CODE REVIEW

File: core/services/bulletprooftxmanager/bulletprooftxmanager.go [hunk 1]
```diff
@@ -380,17 +382,33 @@ func newLegacyTransaction(nonce uint64, to common.Address, value *big.Int, gasLi
...
+	var hash common.Hash
+	hash, err = signedTxHash(signedTx, chainID)
+	if err != nil {
+		return hash, nil, err
+	}
+	return hash, rlp.Bytes(), nil
+}
 
+func signedTxHash(signedTx *gethTypes.Transaction, chainID *big.Int) (hash common.Hash, err error) {
+	if evmtypes.IsExChain(chainID) {
```
Comment (spooktheducks) [inline core/services/bulletprooftxmanager/bulletprooftxmanager.go:403]: Can we inject a `TxHasher` instead? The less refactoring we need to do for CC, the better

**Review by spooktheducks — DISMISSED** [review]

File: core/services/bulletprooftxmanager/bulletprooftxmanager.go (hunk 1 above)
Comment (samsondav) [reply core/services/bulletprooftxmanager/bulletprooftxmanager.go:403]: When multichain is merged, I intend to refactor this to put it as a method on the chain object.

Until then, it makes it easier to refactor to do it like this.

Comment (samsondav) [reply core/services/bulletprooftxmanager/bulletprooftxmanager.go:403]: See: https://app.clubhouse.io/chainlinklabs/story/15991/refactor-okex-special-signing-logic-using-a-txhasher-or-similar

**Review by PiotrTrzpil — DISMISSED** [review]

**Review by PiotrTrzpil — APPROVED** [review]
LGTM

**Pull request merged**